test/integration/capacity_scheduling_test.go

```diff
@@ -0,0 +1,518 @@
+
+	if err = wait.Poll(100*time.Millisecond, 3*time.Second, func() (done bool, err error) {
+		groupList, _, err := cs.ServerGroupsAndResources()
+		if err != nil {
+			return false, nil
+		}
+		for _, group := range groupList {
+			if group.Name == scheduling.GroupName {
+				return true, nil
+			}
+		}
+		t.Log("waiting for crd api ready")
+		return false, nil
+	}); err != nil {
+		t.Fatalf("Waiting for crd read time out: %v", err)
+	}
+
+	cfg := &scheconfig.CapacitySchedulingArgs{
+		KubeConfigPath: kubeConfigPath,
+	}
+
+	profile := schedapi.KubeSchedulerProfile{
+		SchedulerName: v1.DefaultSchedulerName,
+		Plugins: &schedapi.Plugins{
+			PreFilter: &schedapi.PluginSet{
+				Enabled: []schedapi.Plugin{
+					{Name: capacityscheduling.Name},
+				},
+			},
+			PostFilter: &schedapi.PluginSet{
+				Enabled: []schedapi.Plugin{
+					{Name: capacityscheduling.Name},
+				},
+				Disabled: []schedapi.Plugin{
+					{Name: "*"},
+				},
+			},
+			Reserve: &schedapi.PluginSet{
+				Enabled: []schedapi.Plugin{
+					{Name: capacityscheduling.Name},
+				},
+			},
+		},
+		PluginConfig: []schedapi.PluginConfig{
+			{
+				Name: capacityscheduling.Name,
+				Args: cfg,
+			},
+		},
+	}
+
+	testCtx.ClientSet = cs
+	testCtx = util.InitTestSchedulerWithOptions(
+		t,
+		testCtx,
+		true,
+		scheduler.WithProfiles(profile),
+		scheduler.WithFrameworkOutOfTreeRegistry(registry),
+	)
+	t.Log("init scheduler success")
+	defer testutil.CleanupTest(t, testCtx)
+
+	// Create a Node.
+	nodeName := "fake-node"
+	node := st.MakeNode().Name("fake-node").Label("node", nodeName).Obj()
+	node.Status.Allocatable = v1.ResourceList{
+		v1.ResourcePods:   *resource.NewQuantity(32, resource.DecimalSI),
+		v1.ResourceMemory: *resource.NewQuantity(200, resource.DecimalSI),
+		v1.ResourceCPU:    *resource.NewQuantity(200, resource.DecimalSI),
+	}
+	node.Status.Capacity = v1.ResourceList{
+		v1.ResourcePods:   *resource.NewQuantity(32, resource.DecimalSI),
+		v1.ResourceMemory: *resource.NewQuantity(200, resource.DecimalSI),
+		v1.ResourceCPU:    *resource.NewQuantity(200, resource.DecimalSI),
+	}
+	node, err = cs.CoreV1().Nodes().Create(ctx, node, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Failed to create Node %q: %v", nodeName, err)
+	}
+
+	namespaces := []string{"ns1", "ns2"}
+	for _, ns := range namespaces {
+		_, err := cs.CoreV1().Namespaces().Create(ctx, &v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{Name: ns}}, metav1.CreateOptions{})
+		if err != nil && !errors.IsAlreadyExists(err) {
+			t.Fatalf("Failed to integration test ns: %v", err)
+		}
+		autoCreate := false
+		t.Logf("namespaces %+v", ns)
+		_, err = cs.CoreV1().ServiceAccounts(ns).Create(ctx, &v1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{Name: "default", Namespace: ns}, AutomountServiceAccountToken: &autoCreate}, metav1.CreateOptions{})
+		if err != nil && !errors.IsAlreadyExists(err) {
+			t.Fatalf("Failed to create ns default: %v", err)
+		}
+	}
+
+	for _, tt := range []struct {
```

Comment:
Can you add a couple of cases?

1. EQ1 overused, but preempting pods in EQ1 doesn't help. (this is possible is preemptor requests a huge amount of resources)
1. EQ1 not overused, but pods in other over-used EQ have a higher priority, and hence preemptor is not able to preempt.
1. Pods spread in multiple nodes (it's ok to showcase that the current implementation doesn't support cross-node preemption)

Reply:
Done